### ISSUE TITLE
Disable default rules in python

### ISSUE THREAD

**Issue by suresh26k** [aws-cloudformation/cfn-lint#2013]
*cfn-lint version:*  0.49.1

*Description of issue.*

Is there any way to disable all the default rules while using `cfn-lint` with python. We would like our templates to evaluate for only a few custom rules.

**Comment by benbridts** [comment]
CloudFormation Guard might be a better fit for custom rule checking: https://github.com/aws-cloudformation/cloudformation-guard

to answer the question:

```
# -i: ignore
# -a: append your own rules
cfn-lint -i E -i W -i I -a your_rules
```

**Comment by kddejong** [comment]
@benbridts provided the type of command you should be looking at.

Additionally you can also use `-m, --mandatory-checks`.  This allows you to override the affects of the Ignore.  Your use case is a little tricky from how we designed it.  We automatically include EXXXX and WXXXX rules by default. You would also want to look at your rule naming as the the `-i` and `-m` validate using a string starts with logic.  That's why you can ignore all EXXXX rules with `-i E` so if for some reason even after appending rules you aren't getting errors add in `-m E9XXX` if your rules IDs start with `E9`

**Comment by kddejong** [comment]
@suresh26k I can re-open if needed but I think we got you some good answers here.  Let us know if it doesn't work out and thanks for reaching out.

**Comment by suresh26k** [comment]
Thank you @benbridts @PatMyron  for your response.

Following worked for me

* imports
  ```python
  import cfnlint.core
  from cfnlint.rules import RulesCollection
  ```
* Create empty RuleCollection
  ```python
       # We are creating empty rules from RulesCollection
        rules = RulesCollection(
            ignore_rules=[],
            include_rules=[],
            configure_rules=[],
            include_experimental=False,
            mandatory_rules=[],
        )
  ```
* Read rules from rules file (conditional rules)
  *  Code
     ```python
       # Add conditional rules
        rules.create_from_custom_rules_file(
            custom_conditional_rules_file
        )
     ```
  * Rules in this custom conditional rules file are defined like
     ```
     AWS::S3::Bucket AccessControl != PublicRead ERROR "My Custom Error"
     AWS::S3::Bucket AccessControl NOT_EQUALS PublicRead ERROR "AccessControl_PublicRead - Public read should not be enabled"
     ```

* Read custom rules
  * Code
     ```python
      # Reads custom code from a directory/folder
       rules.create_from_directory(custom_rules_dir)
      ```
  * Here, we can create rules as per instructions provided [here](https://github.com/aws-cloudformation/cfn-lint/blob/master/docs/getting_started/rules.md#creating-a-new-rule).
  * Here's an example.
    ```python
    from cfnlint.rules import CloudFormationLintRule
    from cfnlint.rules import RuleMatch
    
    
    class S3ACLPublicReadRule(CloudFormationLintRule):
        """Rule description """
        id = 'EMYRULE123'
        # NOTE: Rule name should begin with 'E' or else it will not be checked.
        shortdesc = 'Checks if public read is enabled for s3 bucket'
        description = 'Checks if public read is enabled for s3 bucket'
        source_url = ''  # A url to the source of the rule, e.g. documentation, AWS Blog posts etc
        tags = []  # A set of tags (strings) for searching
    
        def match(self, cfn):
            """Basic Rule Matching"""
    
            # Get all S3 buckets from the template
            resources = cfn.get_resources(['AWS::S3::Bucket'])
    
            # Error Matches
            matches = []
    
            # Your Rule code goes here
            for resource_name, resource in resources.items():
    
                # Path
                path = ['Resources', resource_name, 'Properties']
    
                # Resource Properties
                properties = resource.get('Properties')
    
                if properties:
                    for key in properties:
                        if key == "AccessControl" and properties[key] == "PublicRead":
                            matches.append(RuleMatch(path, "Public read is enabled"))
    
            # An error will be appended to `matches` if the condition defined here match
            return matches

    ```

* Now match rules
    * Code
       ```python
              matches = cfnlint.core.run_checks(
                  filename=template_file,
                  template=template,
                  rules=rules,
                  regions=regions,
              )
       ```
  * Here
    * template_file => Path of CloudFormation template
    * template => CloudFormation template
      * It is of type `<class 'cfnlint.decode.node.create_dict_node_class.<locals>.node_class'>`
      * We can get it using following for json file - `template = cfnlint.decode.cfn_json.load(template_file)`
    * rules => Rules which we fetched intially
    * regions => list of regions. like `["us-east-2"]`
